src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -201,11 +146,18 @@ object FilterIndexRule
    * @return top-most index which is expected to maximize performance gain
    *         according to ranking algorithm.
    */
-  private def rank(candidates: Seq[IndexLogEntry]): Option[IndexLogEntry] = {
+  private def rank(
+      candidates: Seq[IndexLogEntry],
+      hybridScanEnabled: Boolean = false): Option[IndexLogEntry] = {
     // TODO: Add ranking algorithm to sort candidates.
     candidates match {
       case Nil => None
-      case _ => Some(candidates.head)
+      case _ =>
+        if (hybridScanEnabled) {
+          Some(candidates.maxBy(_.allSourceFiles.size))
```

Comment:
This is basically saying, `pick the index with the largest number of buckets`, right?

Reply:
It's `pick the index with the largest number of source files` of an index. (just for append-only)

Reply:
Got it. Isn't source file the same as bucket files for an index?

Reply:
It's not the number of index data files, but the number of source data files in the source relation.

Reply:
I think the best way I understood this is: `pick the index that indexes (or covers) the most source data (determined by allSourceFiles captured as part of the index)`